powerauth-restful-security-base/src/main/java/io/getlime/security/powerauth/rest/api/base/encryption/PowerAuthEciesEncryption.java

```diff
@@ -37,6 +42,36 @@
     private EciesDecryptor eciesDecryptor;
     private byte[] encryptedRequest;
     private byte[] decryptedRequest;
+    private T requestObject;
+
+    /**
+     * Initialize ECIES encryption object from PowerAuth encryption HTTP header.
+     *
+     * @param encryptionHttpHeader PowerAuth encryption HTTP header.
+     * @throws PowerAuthEncryptionException In case PowerAuth encryption HTTP header is invalid.
+     */
+    public PowerAuthEciesEncryption(String encryptionHttpHeader) throws PowerAuthEncryptionException {
```

Comment:
In case the signature is present alongside the encryption, there is no encryption header, only the signature header is present. Therefore, this instance must be very likely initialized using something else...

Reply:
You are right, I will introduce a more generic constructor.